pkg/migration/plan_executor.go

```diff
@@ -103,14 +111,15 @@ func NewPlanExecutor(plan Plan, executors []Executor, opts ...PlanExecutorOption
 
 func (pe *PlanExecutor) Execute() error { //nolint:gocyclo // easier to follow this way
 	ctx := make(map[string]any)
-	for i := 0; i < len(pe.plan.Spec.Steps); i++ {
+	for i := pe.StartIndex; i < len(pe.plan.Spec.Steps); i++ {
```

Comment:
Do we have access to a logger in `migration.PlanExecutor`? If so, we may consider adding at least a debug statement that will convey the step index we are starting execution at.

Reply:
We don't have access to a logger. One option can be adding a log to callback implementation.